docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,428 @@
+    "Submit a new vote"
+    (enforce (contains key ["A" "B"]) "You can only vote for A or B")
+    (with-read votes key { "count" := count }
+      (update votes key { "count": (+ count 1) })
+    )
+    (format "Voted {}!" [key])
+  )
+
+  (defun getVotes:integer (key:string)
+    "Get the votes count by key"
+    (at 'count (read votes key ['count]))
+  )
+
+  (defun init ()
+    "Initialize the rows in votes table"
+    (insert votes "A" { "option": "A", "count" : 0 })
+    (insert votes "B" { "option": "B", "count" : 0 })
+  )
+)
+
+(create-table votes)
+(init)
+```
+
+Notice how we're calling `create-table` and `init` outside of our module. That's an important step, so don't miss it.
+
+We've got our module logic so now let's add a few more tests to make sure everything is right. Open the `vote.repl` file and copy the following snippet:
+
+```clojure
+(begin-tx)
+(use simple-vote)
+(expect-failure "Vote fails if C" (vote "C"))
+(commit-tx)
+
+(begin-tx)
+
+;; import simple-vote module
+(use simple-vote)
+;; test if votes count for key "A" is initialized with 0
+(expect "votes for A initialized" (getVotes "A") 0)
+
+;; test if votes count for key "B" is initialized with 0
+(expect "votes for B initialized" (getVotes "B") 0)
+
+(commit-tx)
+
+(begin-tx)
+(use simple-vote)
+
+;; test if votes count for key "A" is correctly increased by 1
+(let ((count (getVotes "A")))
+  (vote "A")
+  (expect "votes count is increased by 1" (getVotes "A") (+ count 1)))
+
+;; test if votes count for key "B" is correctly increased by 1
+(let ((count (getVotes "B")))
+  (vote "B")
+  (expect "votes count is increased by 1" (getVotes "B") (+ count 1)))
+
+(commit-tx)
+```
+Notice how we are using `expect` to test if `getVotes` returns the correct result.
+`expect` is one the functions that's only available in the REPL but there are many more, check them out <a href="https://pact-language.readthedocs.io/en/latest/pact-functions.html#repl-only-functions">here</a>.
+
+I also want to mention the `let` construct which is helpful when you need to bind some variables be in the same scope as some other logic that makes use of them. In our case we first loaded the number of votes and binded the result to `count` variable which we used later in the `expect` function.
+
+You can read more about `let` and `let*` <a href="https://pact-language.readthedocs.io/en/latest/pact-reference.html#let"> here</a>.
+
+Let's run those tests again to make sure everything works as expected:
+
+```clojure
+$ pact
+pact> (load "vote.repl")
+```
+
+
+### Gas Station
+
+A unique feature of Kadena is the ability to allow gas to be paid by a different entity than the one who initiated the transaction. This entity is what we call a "gas station".
+
+Let's create our own gas station that will allow users to submit votes without paying for gas, instead gas will be subsidized by the gas station.
+
+Each gas station needs to implement the `gas-payer-v1` interface which is shown below:
```

Comment:
Use this opportunity to explain what an interface is, and that you can find this particular one on chain X (all of them? )

Reply:
There's an info section just below the snippet that explain it.

Reply:
IMHO this shouldn't be part of this documentation but part of another section that we can link to. We implement a gas station so we can tell why we create a gas station but not what a gas station is and does. I do like the text we should just move it.